lib/src/execution/windows/execution.cc

```diff
@@ -358,6 +358,24 @@ namespace facter { namespace execution {
         option_set<execution_options> const& options,
         uint32_t timeout)
     {
+        // Since we use a job object in the windows world, we want to
+        // be sure we're not in a job object, or at least able to
+        // break our processes out if we are in one.
+        BOOL in_job;
+        bool use_job_object = true;
+        if (!IsProcessInJob(GetCurrentProcess(), nullptr, &in_job)) {
+            throw execution_exception("could not determine if facter is in a object");
+        }
+        if (in_job) {
+            JOBOBJECT_BASIC_LIMIT_INFORMATION limits;
+            if (!QueryInformationJobObject(nullptr, JobObjectBasicLimitInformation, &limits, sizeof(limits), nullptr)) {
+                throw execution_exception("could not query the job object we are running under");
```

Comment:
Could this call fail because the job doesn't have `JOB_OBJECT_QUERY` set?  Perhaps if it does fail, instead of throwing, we just assume we shouldn't create a job?